corehq/form_processor/utils/xform.py

```diff
@@ -51,7 +51,7 @@
     ([0-5]\d)?  # second
     \D?
     (\d{3,6})?  # millisecond
-    ([zZ]|([\+-])([01]\d|2[0-3])\D?([0-5]\d)?)  # timezone
+    ([zZ]|([\+-])([01]\d|2[0-3])\D?([0-5]\d)?)?  # timezone
```

Comment:
Did this undo the original intent of this PR? Was that intentional?

Reply:
Yes, I was going to update after the tests passed, but synced with @dannyroberts and it seems like this change would cause drastic behavior that is too unexpected for users in how datetime strings are represented. A better approach might be to provide another isolated solution (like another function) that does not affect all users.

I thought to revert it the functionality but keep the refactoring if you prefer this organization, but I can also close the PR as well.

Reply:
I like the refactor. Let's keep it.